starknet_py/net/account/account_client.py

```diff
@@ -55,39 +55,44 @@ def __init__(
         client: Client,
         signer: Optional[BaseSigner] = None,
         key_pair: Optional[KeyPair] = None,
+        chain: Optional[StarknetChainId] = None,
     ):
         """
         :param address: Address of the account contract
         :param client: Instance of GatewayClient which will be used to add transactions
         :param signer: Custom signer to be used by AccountClient.
-                       If none is provieded, default
+                       If none is provided, default
                        :py:class:`starknet_py.net.signer.stark_curve_signer.StarkCurveSigner` is used.
         :param key_pair: Key pair that will be used to create a default `Signer`
+        :param chain: ChainId of the chain used by the client
```

Comment:
Perhaps here we should do a warning like `Chain will be a required parameter in the next version.`?

Reply:
Actually, the chain won't be a required parameter. It will be optional, but chain or signer must be provided and we have an exception for that.